params/config.go

```diff
@@ -540,7 +540,12 @@ func (c *ChainConfig) String() string {
 		engine = "unknown"
 	}
 
-	return fmt.Sprintf("{ChainID: %v Homestead: %v DAO: %v DAOSupport: %v EIP150: %v EIP155: %v EIP158: %v Byzantium: %v Constantinople: %v Petersburg: %v Istanbul: %v, Muir Glacier: %v, Ramanujan: %v, Niels: %v, MirrorSync: %v, Bruno: %v, Berlin: %v, YOLO v3: %v, CatalystBlock: %v, London: %v, ArrowGlacier: %v, MergeFork:%v, Euler: %v, Gibbs: %v, Nano: %v, Moran: %v, Planck: %v,Luban: %v, Plato: %v, Hertz: %v, Engine: %v}",
+	var ShanghaiTime *big.Int
+	if c.ShanghaiTime != nil {
+		ShanghaiTime = big.NewInt(0).SetUint64(*c.ShanghaiTime)
```

Comment:
this can be simplified to `ShanghaiTime.SetUint64(*c.ShanghaiTime)` or `ShanghaiTime = ShanghaiTime.SetUint64(*c.ShanghaiTime)`

Reply:
> this can be simplified to `ShanghaiTime.SetUint64(*c.ShanghaiTime)` or `ShanghaiTime = ShanghaiTime.SetUint64(*c.ShanghaiTime)`

`ShanghaiTime` is nil before line 544

Reply:
correct, my bad